src/libraries/Common/src/System/Net/SocketAddressPal.Unix.cs

```diff
@@ -165,5 +158,13 @@ public static unsafe void SetIPv6Address(byte[] buffer, byte* address, int addre
 
             ThrowOnFailure(err);
         }
+
+        public static unsafe void Clear(Span<byte> buffer)
+        {
+            AddressFamily family = GetAddressFamily(buffer);
+            buffer.Clear();
+            buffer[0] = (byte)buffer.Length;
```

Comment:
This requires that the buffer length never be > 255?  Do we need an assert for that somewhere?

Reply:
This is really hack applicable to macOS-like and BSD* Where the length is part of the socket address. 
https://gist.github.com/cyberroadie/3490843#file-socket-h-L326-L330
It was added because new tests were failing Equals() in the test. (probably existing gap) 

On Windows & Linux this will get overeaten by the subsequent set of address family and the limitation is not applicable.